src/wallet/scriptpubkeyman.cpp

```diff
@@ -208,6 +208,47 @@ IsMineResult IsMineInner(const LegacyScriptPubKeyMan& keystore, const CScript& s
 
 } // namespace
 
+SigningResult ScriptPubKeyMan::SignMessageBIP322(MessageSignatureFormat format, const SigningProvider* keystore, const std::string& message, const CTxDestination& address, std::string& str_sig) const
+{
+    assert(format != MessageSignatureFormat::LEGACY);
+
+    MessageVerificationResult result; // unused
+    auto txs = BIP322Txs::Create(address, message, result);
+    if (!txs) return SigningResult::PRIVATE_KEY_NOT_AVAILABLE;
```

Comment:
In f0747683eb73f2052b9ef18b8246e81411c30e20 "wallet: add SignMessageBIP322() helper method to ScriptPubKeyMan"

This could be an assert. It should not be possible to reach this point without having a valid destination, and as we do not provide a signature argument, `Create` should never fail.

Reply:
I'm not sure that's the case - what if `address` does not contain a destination?

Reply:
It shouldn't. First, both the RPC and the GUI will validate that `address` is a valid destination before passing it on. Then `CWallet::SignMessage` also ensures that there is a valid destination by getting the script for the destination and using the `ScriptPubKeyMan` that can provide for that script. So by the time we get to this line, we've already checked at least twice that the destination is valid.

Reply:
Switched to asserting.